crates/wasmtime/src/engine.rs

```diff
@@ -239,6 +239,20 @@ impl Engine {
         Ok(mmap.to_vec())
     }
 
+    /// Returns a value that can be used to check precompiled WebAssembly compatibility.
+    ///
+    /// The outputs of [`Engine::precompile_module`] and [`Engine::precompile_component`]
+    /// are compatible with a different [`Engine`] instance only if the two engines use
+    /// compatible [`Config`]s. The value returned by this function will be the same for two
+    /// [`Engine`]s if a binary from one can be deserialized by the other.
+    #[cfg(compiler)]
+    #[cfg_attr(nightlydoc, doc(cfg(feature = "cranelift")))] // see build.rs
+    pub fn precompile_compatibility_key(&self) -> impl AsRef<[u8]> {
+        use sha2::Digest;
+        let data = serialization::build_compiler_info(&self);
+        sha2::Sha256::digest(data)
```

Comment:
sha512 is faster on 64bit systems as it can consume more bytes at once. Not that it really matters here. This hashing is highly unlikely to ever become a bottleneck.